lib/ex_plasma/transaction/type/payment_v1.ex

```diff
@@ -1,150 +1,249 @@
+  """
+  @spec new(outputs(), outputs(), metadata()) :: t()
+  def new(inputs, outputs, metadata) do
+    %__MODULE__{tx_type: @tx_type, inputs: inputs, outputs: outputs, tx_data: @empty_tx_data, metadata: metadata}
+  end
+
+  @spec new(outputs(), outputs()) :: t()
+  def new(inputs, outputs), do: new(inputs, outputs, @empty_metadata)
 
   @doc """
-  Encode the given Transaction into an RLP encodeable list.
+  Creates output for a payment v1 transaction
 
   ## Example
 
-  iex> txn = %ExPlasma.Transaction{
-  ...>  inputs: [%ExPlasma.Output{output_data: nil, output_id: %{blknum: 0, oindex: 0, position: 0, txindex: 0}, output_type: nil}],
-  ...>  metadata: <<0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0>>,
-  ...>  outputs: [
-  ...>    %ExPlasma.Output{
-  ...>      output_data: %{amount: 1, output_guard: <<29, 246, 47, 41, 27, 46, 150, 159, 176, 132, 157, 153, 217, 206, 65, 226, 241, 55, 0, 110>>,
-  ...>        token: <<46, 38, 45, 41, 28, 46, 150, 159, 176, 132, 157, 153, 217, 206, 65, 226, 241, 55, 0, 110>>}, output_id: nil, output_type: 1
-  ...>    }
-  ...>  ],
-  ...>  sigs: [],
-  ...>  tx_data: <<0>>,
-  ...>  tx_type: 1
-  ...>}
-  iex> ExPlasma.Transaction.Type.PaymentV1.to_rlp(txn)
-  [[], <<1>>, [<<0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0>>], [[<<1>>, [<<29, 246, 47, 41, 27, 46, 150, 159, 176, 132, 157, 153, 217, 206, 65, 226, 241, 55, 0, 110>>, <<46, 38, 45, 41, 28, 46, 150, 159, 176, 132, 157, 153, 217, 206, 65, 226, 241, 55, 0, 110>>, <<1>>]]], 0, <<0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0>>]
+  iex> output = new_output(<<1::160>>, <<0::160>>, 1)
+  iex> %ExPlasma.Output{
+  ...>   output_data: %{amount: 1, output_guard: <<1::160>>, token: <<0::160>>},
+  ...>   output_id: nil,
+  ...>   output_type: 1
+  ...> } = output
+  """
+  @spec new_output(Crypto.address_t(), Crypto.address_t(), pos_integer()) :: Output.t()
+  def new_output(owner, token, amount) do
+    %Output{
+      output_type: @output_type,
+      output_data: %{
+        amount: amount,
+        output_guard: owner,
+        token: token
+      }
+    }
+  end
+end
+
+defimpl ExPlasma.Transaction.Protocol, for: ExPlasma.Transaction.Type.PaymentV1 do
+  alias ExPlasma.Output
+  alias ExPlasma.Transaction.TypeMapper
+  alias ExPlasma.Transaction.Type.PaymentV1
+  alias ExPlasma.Utils.RlpDecoder
+
+  @empty_metadata <<0::256>>
+  @empty_tx_data 0
+  @output_limit 4
+
+  @tx_type TypeMapper.tx_type_for(:tx_payment_v1)
+  @output_type TypeMapper.output_type_for(:output_payment_v1)
+
+  @type validation_error() ::
+          {:inputs, :duplicate_inputs}
+          | {:inputs | :outputs, :cannot_exceed_maximum_value}
+          | {:inputs | :outputs, :cannot_subceed_minimum_value}
+          | {:outputs, :invalid_output_type_for_transaction}
+          | {:tx_data, :malformed_tx_data}
+          | {:metadata, :malformed_metadata}
+          | {atom(), atom()}
```

Comment:
do we want to leave this? it seems like there was effort to define all validation errors, but this leaves "other"